### DISCUSSION TITLE
Can I specify a supported version of the Pkl language?

### DISCUSSION THREAD

**Discussion by fumito-ito** [apple/pkl#274] [Q&A] [answered]
Hi team !

I have just released a module to generate type-safe `xcprivacy` files using the Pkl language.
https://github.com/fumito-ito/PrivacyManifest

I would like to express the version of the Pkl language supported by this module in the code, is there a proper way to do this? I often see expressions like `@ModuleInfo { minPklVersion = "0.25.0" }` when referring to pkl-pantry, should I use this?

**Comment by bioball** [comment, marked as answer]
Hi, yes, that annotation is how you express the version of the Pkl language supported by your code.

And, awesome! This looks pretty good!

I'd recommend setting an output renderer in your template.

```
output {
  renderer = new PListRenderer {}
}
```

That way, you can avoid needing to use the `-f` flag when rendering.

It also means that someone can use your module and render it to plist just calling `output.text`, regardless of whether the `-f` flag was set or not.

**Reply by fumito-ito** [reply]
Thanks @bioball ! I fixed it !
